examples/openai-lander/evolve.py

```diff
@@ -235,21 +234,22 @@ def run():
 
             if solved:
                 print("Solved.")
-
-                # Save the winners.
-                for n, g in enumerate(best_genomes):
-                    name = 'winner-{0}'.format(n)
-                    with open(name + '.pickle', 'wb') as f:
-                        pickle.dump(g, f)
-
-                    visualize.draw_net(config, g, view=False, filename=name + "-net.gv")
-                    visualize.draw_net(config, g, view=False, filename=name + "-net-pruned.gv", prune_unused=True)
-
                 break
         except KeyboardInterrupt:
             print("User break.")
             break
 
+        finally:
+            # Save the winners.
```

Comment:
You could put an `if best_genomes:` here to handle the case where the user terminates within the first 5 generations (if I read it correctly, `best_genomes` will be undefined in that case).

Reply:
good point. I defined `best_genomes = None` in the top to make sure the variable exists and can be checked and added an `if best_genomes` on the bottom